packages/plugin-apex/src/commands/force/apex/test/run.ts

```diff
@@ -374,4 +357,23 @@ export default class Run extends SfdxCommand {
     const hint = messages.getMessage('apexTestReportFormatHint', [reportArgs]);
     return hint;
   }
+
+  /** Handles special exceptions where we don't want to return early
+   * with the testRunId.*/
+  private shouldExitEarly(): boolean {
```

Comment:
This function addresses a code smell where we were conflating the meaning of two variables; previously we had resultformat != undefined to equal !exitOnTestRunId, which was causing some confusion further down the stack. Now we're calling out specific circumstances in which we should/should not exit early. The logic is a little inversed here, but I prefer "should" to "should not" variables as an easier way to keep variables straight.